src/graph.jl

```diff
@@ -0,0 +1,349 @@
+using DataFrames
+using GraphDataFrameBridge
+using MetaGraphs
+using CSV
+using LightGraphs
+using GraphPlot
+using Random
+using DataFramesMeta
+
+function load_graph_data(input_data)
+    
+    return input_data
+end
+    
+"""    generate_synthetic_vertices(vertex_type_defs::String, output_path::String)
+
+Generate synthetic test data. The synthetic vertices are returned as a dataframe 
+that can be used for testing/debugging/developing the knowledge graph.
+
+see also: generate_synthetic_edges
+"""
+function generate_synthetic_vertices(vertex_type_defs::String, output_path::String)
+
+    
+    vertex_types = CSV.read(vertex_type_defs)
+    
+    synth_vertices_df = []
+    
+    for vertex in eachrow(vertex_types)
+        
+        # Generate a synthetic name; hash the concatenation of name||type to get a unique vertex id
+        v_name = string("synth_", vertex.v_type, "_", randstring(MersenneTwister(), 'a':'z', 6)) 
+        v_hash = hash(string(v_name, vertex.v_type))       
```

Comment:
I think we should use a function called `genname` to create new vertex names.  There is a `Base` function `gensym` https://docs.julialang.org/en/v1.0/base/base/#Base.gensym, which makes a new symbol name that is guaranteed to be unique. The symbols are basically the `tag` which is provided as an argument and then some `#` and numbers.

```julia
julia> gensym(:banana)
Symbol("##banana#360")
```

We could actually use gensym and say that vertex names are symbols? I don't know if that would be better or worse than using strings.

Reply:
I'm fine with this; the only potential drawback I see is usability/searchability, etc. I thought the name would be as close to natural language as possible (so that the users can find/understand vertices?) and then the hash would be a guaranteed unique identifier. @jpfairbanks , do you want to use gensym for in addition to name/hash, or in lieu of one or both of them?

Reply:
Yeah I do want the name to be close to human readable "def_func_x123" is more readable than "0x1238a0c0bf" right? 

I guess this code does two steps, one generate a random vertex name, two hash that name to create a vertex id which is uint64.

For step one I think we can go with making a function: `randname(vertex, tag="synth") = string(tag, vertex.v_type, "_", randstring(MersenneTwister(), 'a':'z', 6))`.

For step two, we should extract out that logic into a subroutine that will be easier to change later. 
`vid(name, type) = hash(string(name, type))`

Reply:
made subroutines for these.